pyproject.toml

```diff
@@ -6,3 +6,118 @@ profile = "black"
 line_length = 79
 # We patch logging in main.py before certain imports
 skip = ["cloudinit/cmd/main.py", ".tox", "packages", "tools"]
+
+[tool.mypy]
+follow_imports = "silent"
+exclude=[
+  '^cloudinit/apport\.py$',
+  '^cloudinit/cmd/query\.py$',
+  '^cloudinit/config/cc_apk_configure\.py$',
```

Comment:
I like the direction and intent. Given the size of the excludes list, I'd want to tackle a followup on this in one big PR after we merge this to avoid thrashing downstreams with multiple format-based paper cut PRs that add up to an annoyance over a period of time. Given that we have already done a fairly big non-functional reformat due to unittest restructuring, black and isort formatting. I wonder if it makes sense for us to chunk this work out and try to squeeze it in sooner rather than later so downstreams who feel that format-based impact for black/isort/unittests only have to cross this bridge with non-functional merge-conflcts in one set of PRs across the 21.4 to 22.1 barrier.

Reply:
This isn't a formatting concern though. This is actual refactoring to ensure types align correctly. Mypy doesn't complain about non-typed stuff. It complains when expected types don't line up, and we have thousands of them. I don't think this is a trivial "do it all in one or two commits" type exercise.